controllers/index.js

```diff
@@ -36,17 +38,48 @@ exports.getArticles = (req, res, next) => {
 exports.getArticleById = (req, res, next) => {
   const { article_id } = req.params;
   if (!/^\d+$/.test(article_id)) {
-    res.status(400).send({ msg: "invalid id" });
+    res.status(400).send({ message: "Invalid ID" });
   } else {
     fetchArticleById(article_id)
-      .then(({ rows }) => {
+      .then((article) => {
+        res.status(200).send({ article });
+      })
+      .catch(next);
+  }
+};
+
+exports.getComments = (req, res, next) => {
+  const { article_id } = req.params;
+  if (!/^\d+$/.test(article_id)) {
+    res.status(400).send({ message: "Invalid ID" });
+  } else {
+    Promise.all([fetchComments(article_id), fetchArticleById(article_id)])
+      .then(([{ rows }, result2]) => {
         if (rows.length === 0) {
-          res.status(404).send({ message: "article not found" });
+          res.status(200).send({ message: "No comments", comments: [] });
         } else {
-          const article = rows[0];
-          res.status(200).send({ article });
+          rows.article_id = +article_id;
+          res.status(200).send({ comments: rows });

```

Comment:
There is no need to have two seperate responses being sent this way, in either case it should be the same response being sent through.